src/keywords/enum_.rs

```diff
@@ -53,5 +144,36 @@ pub fn compile(
     schema: &Value,
     _: &CompilationContext,
 ) -> Option<CompilationResult> {
-    Some(EnumValidator::compile(schema))
+    if let Value::Array(items) = schema {
+        let mut types = PrimitiveTypesBitMap::new();
+        let mut numbers = vec![];
+        let mut strings = vec![];
+        for item in items {
+            match item {
+                Value::Array(_) => types |= PrimitiveType::Array,
```

Comment:
I think it is possible to exit this loop somehow early if a different type occured